### PR TITLE
Add option to disable salt execution

### PR DESCRIPTION
**Pull request by arbulu89**
A potential user has requested the option to disable the salt provisioning in order to apply custom options. 
This PR adds the option to disable salt execution.
Besides that, I have structured the code to add easily other provisioning methods (ansible for example).

By now, I have only added the option to Libvirt. If you agree with the changes I will extend the change to the other providers following a similar structure.

Actually, in the future, we could think about unifying all the provisioning for all the customers in a unique module, but this requests more changes.

### CODE REVIEW

**Comment by arbulu89** [comment]
One small improvement would be to replace the `enable_salt` option to `provisioner` (or similar) this being a string with available options like `salt`, `ansible`, etc. It will reduce the used variables amount

**Comment by diegoakechi** [comment]
> One small improvement would be to replace the `enable_salt` option to `provisioner` (or similar) this being a string with available options like `salt`, `ansible`, etc. It will reduce the used variables amount.
> 
> What do you think?

I like this more. Fewer variables to manage. It also allows to have custom provision, as the user could set this variable to their own customized provisioning tf file.

**Comment by arbulu89** [comment]
Azure and AWS updated too.
Now playing with `provisioner` and `background` options in the tfvars file we can modify the execution

**Comment by juadk** [comment]
@arbulu89 @diegoakechi, I'll do a review asap but please, don't merged it this week since we have the GMC build validation.

**Comment by arbulu89** [comment]
Return code issues fixed. Now when any salt deployment fails the terraform execution will fail (but only the provisioner resource, so the VMs will be correctly created)

File: aws/terraform_salt/init-server.tpl [hunk 1]
```diff
@@ -1,27 +0,0 @@
...
-# Node servers are deregistered just after
-# If you don't use qa_mode, a complete registration is done by salt later
-
-if grep -q 'role: "iscsi_srv"' /tmp/grains; then
-  sh /root/salt/install-salt-minion.sh -r ${regcode}
-elif [[ ! -e /usr/bin/salt-minion ]]; then
-  sh /root/salt/install-salt-minion.sh -d -r ${regcode}
-fi
- 
-mkdir -p /etc/salt; mv /tmp/grains /etc/salt/
```
Comment (juadk) [inline aws/terraform_salt/init-server.tpl]: I tested AWS and nothing move /tmp/grains to /etc/salt anymore so the salt deployment fails.

**Comment by arbulu89** [comment]
> AWS, Azure and libvirtd have been modified to support the new variables, but GCP is missing...

GCP updated

File: aws/terraform_salt/init-server.tpl (hunk 1 above)
Comment (arbulu89) [reply aws/terraform_salt/init-server.tpl]: Issue fixed in the last commit

**Comment by ldevulder** [comment]
I tested GCP with success and @juadk  also tested on AWS, Azure and GCP.

We didn't check on libvirt, but not first target, so we could modify later if needed.

**Pull request merged**